nipype/pipeline/engine/auxiliary.py

```diff
@@ -0,0 +1,206 @@
+import pdb
+import inspect
+from ... import config, logging
+logger = logging.getLogger('nipype.workflow')
+
+
+# Function to change user provided mapper to "reverse polish notation" used in State
+def mapper2rpn(mapper):
+    """ Functions that translate mapper to "reverse polish notation."""
+    global output_mapper
```

Comment:
I used an recursion algorithm and using global seemed to be easier solution, but I can think how to avoid it.

In general this file has some extra functions that I'm using, probably they can be part of other classes, if we don't want to have an extra file.

The algorithm just allows to represent a mapper in a way that can be easily read (and iterate) from left to right, so this representation can be much easier to use in other functions (like mapping specific inputs to specific axes). This should be completely private function, users don't have to care about it.